ydb/docs/en/core/yql/reference/types/serial.md

```diff
@@ -0,0 +1,95 @@
+| Serial      | 2^31–1                | Int32    |
+| Serial4     | 2^31–1                | Int32    |
+| Serial8     | 2^63–1                | Int64    |
+| BigSerial   | 2^63–1                | Int64    |
+
+If the sequence reaches its maximum value, insertion will result in an error:
+
+```text
+Error: Failed to get next val for sequence: /dev/test/users/_serial_column_user_id, status: SCHEME_ERROR
+    <main>: Error: sequence [OwnerId: <some>, LocalPathId: <some>] doesn't have any more values available, code: 200503
+```
+
+{% note info %}
+
+The next value is allocated by the generator before the actual insertion into the table and is considered used even if the row is not successfully inserted (for example, in case of transaction rollback).
+As a result, the values in such a column may have gaps and may not form a continuous sequence.
+
+{% endnote %}
+
+Tables with `Serial` columns support [copy](../../../reference/ydb-cli/tools-copy.md), [rename](../../../reference/ydb-cli/commands/tools/rename.md), [dump](../../../reference/ydb-cli/export-import/tools-dump.md), [restore](../../../reference/ydb-cli/export-import/import-file.md), and [import](../../../reference/ydb-cli/export-import/import-s3.md)/[export](../../../reference/ydb-cli/export-import/export-s3.md) operations.
+
+## Usage Example
+
+You should carefully choose the columns for your [PRIMARY KEY](../../../dev/primary-key/row-oriented.md). For scalability and high performance, you should avoid writing rows with monotonically increasing primary keys. In this case, all records will go to the last partition, and all the load will target a single server.
+
+As a recommended approach, use a hash (for example, from the whole or a part of the primary key) as the first key element, which will help evenly distribute data across cluster partitions.
+
+```yql
+CREATE TABLE users (
+    user_hash Uint64,
+    user_id Serial,
+    name Utf8,
+    email Utf8,
+    PRIMARY KEY (user_hash, user_id)
+);
+```
+
+The `user_hash` field can be calculated on the application side, for example, by applying a hash function to the `email`.
+
+``` yql
+UPSERT INTO users (user_hash, name, email) VALUES (123456789, 'Alice', 'alice@example.com');
+INSERT INTO users (user_hash, name, email) VALUES (987654321, 'Bob', 'bob@example.com');
+REPLACE INTO users (user_hash, name, email) VALUES (111111111, 'John', 'john@example.com');
+```
+
+Result (example `user_hash` values are used):
+
+| user_hash   | email                 | name  | user_id |
+|-------------|-----------------------|-------|---------|
+| 123456789   | `alice@example.com`   | Alice | 1       |
+| 987654321   | `bob@example.com`     | Bob   | 2       |
+| 111111111   | `john@example.com`    | John  | 3       |
+
+You can also explicitly specify a value for the `Serial` column during insertion, for example, when restoring data. In this case, the insertion will work like with a regular integer column, and the `Sequence` will not be affected:
```

Comment:
This belongs to a generic description of inserting into columns with `Serial` type and is not specific to this example.

Also, it sounds like this needs to explicitly say that doing so without adjusting the sequence position is likely to result in a primary key conflict either immediately or at a random moment in the future when the sequence will catch up with the inserted value. Or is it smart enough to transparently step over occupied values?